sequencer/src/enriched_handle.rs

```diff
@@ -0,0 +1,55 @@
+use hotshot::{traits::NodeImplementation, types::SystemContextHandle};
+use hotshot_types::{light_client::StateSignature, traits::node_implementation::NodeType};
+use std::{
+    ops::{Deref, DerefMut},
+    sync::{Arc, RwLock},
+};
+
+use crate::api::signature_pool::StateSignaturePool;
+
+pub struct EnrichedSystemContextHandle<TYPES: NodeType, I: NodeImplementation<TYPES>> {
```

Comment:
Can we think of a better name for this?

Reply:
I'm thinking about calling it `context::SequencerContext` that includes a consensus handle, node index and the collection of state signatures. Then the `SequencerNode` would contains only a sequencer context and the list of update tasks. WDYT?